src/base/QXmppMixIq_p.h

```diff
@@ -40,6 +40,50 @@ class QXMPP_EXPORT QXmppMixSubscriptionUpdateIq : public QXmppIq
     QSharedDataPointer<QXmppMixSubscriptionUpdateIqPrivate> d;
 };
 
+class QXMPP_EXPORT QXmppMixInvitationRequestIq : public QXmppIq
+{
+public:
+    QXmppMixInvitationRequestIq();
+
+    QXMPP_PRIVATE_DECLARE_RULE_OF_SIX(QXmppMixInvitationRequestIq)
+
+    QString inviteeJid() const;
+    void setInviteeJid(const QString &);
+
+    /// \cond
+    static bool isMixInvitationRequestIq(const QDomElement &);
+
+protected:
+    void parseElementFromChild(const QDomElement &) override;
+    void toXmlElementFromChild(QXmlStreamWriter *) const override;
+    /// \endcond
+
+private:
+    QSharedDataPointer<QXmppMixInvitationRequestIqPrivate> d;
```

Comment:
I'd rather remove this d pointer here, as this is a private class and probably will be in the future.

(same with the other private classes)

In the blocking manager I even avoided all of the other boilerplate code and did this: https://github.com/qxmpp-project/qxmpp/blob/master/src/client/QXmppBlockingManager.cpp#L48